Mathlib/FieldTheory/Galois/InfiniteDimensional.lean

```diff
@@ -0,0 +1,207 @@
+
+instance (L₁ L₂ : IntermediateField k K) [Algebra.IsSeparable k L₂] :
+    Algebra.IsSeparable k ↑(L₁ ⊓ L₂) :=
+  .of_algHom _ _ (IntermediateField.inclusion inf_le_right)
+
+instance (L₁ L₂ : IntermediateField k K) [IsGalois k L₁] [IsGalois k L₂] :
+    IsGalois k ↑(L₁ ⊓ L₂) := {}
+
+instance : Sup (FiniteGaloisIntermediateField k K) where
+  sup L₁ L₂ := .mk <| L₁.val ⊔ L₂.val
+
+instance : Inf (FiniteGaloisIntermediateField k K) where
+  inf L₁ L₂ := .mk <| L₁.val ⊓ L₂.val
+
+instance : Lattice (FiniteGaloisIntermediateField k K) :=
+  val_injective.lattice _ (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+instance : OrderBot (FiniteGaloisIntermediateField k K) where
+  bot := .mk ⊥
+  bot_le _ := bot_le (α := IntermediateField _ _)
+
+variable (k) in
+/--Define the finite Galois closure of adjoining a finite set-/
+noncomputable def adjoin [IsGalois k K] (s : Set K) [Finite s] :
+    FiniteGaloisIntermediateField k K where
+  val := normalClosure k (IntermediateField.adjoin k (s : Set K)) K
+  to_finiteDimensional :=
+    letI : FiniteDimensional k (IntermediateField.adjoin k (s : Set K)) :=
+      IntermediateField.finiteDimensional_adjoin <| fun z _ =>
+        IsAlgebraic.isIntegral (Algebra.IsAlgebraic.isAlgebraic z)
+    normalClosure.is_finiteDimensional k (IntermediateField.adjoin k (s : Set K)) K
+  to_isGalois := IsGalois.normalClosure k (IntermediateField.adjoin k (s : Set K)) K
+
+lemma adjoin_val [IsGalois k K] (s : Set K) [Finite s] :
+    (FiniteGaloisIntermediateField.adjoin k s).val =
+    normalClosure k (IntermediateField.adjoin k s) K :=
+  rfl
+
+variable (k) in
+lemma subset_adjoin [IsGalois k K] (s : Set K) [Finite s] :
+    s ⊆ (adjoin k s).val := by
+  intro x hx
+  apply IntermediateField.le_normalClosure
+  simp only [IntermediateField.adjoin, Set.union_insert, Set.union_singleton,
+    IntermediateField.mem_mk, Subring.mem_toSubsemiring, Subfield.mem_toSubring]
+  apply Subfield.subset_closure
+  simp only [Set.mem_union, Set.mem_range, hx, or_true]
+
+@[simp]
+theorem adjoin_le_iff [IsGalois k K] {s : Set K} [Finite s]
+    {L : FiniteGaloisIntermediateField k K} : adjoin k s ≤ L ↔ s ≤ L.val := by
+  show normalClosure _ _ _ ≤ L.val ↔ _
+  rw [← IntermediateField.adjoin_le_iff, IntermediateField.normalClosure_le_iff_of_normal]
+
+theorem adjoin_simple_le_iff [IsGalois k K] {x : K} {L : FiniteGaloisIntermediateField k K} :
+    adjoin k {x} ≤ L ↔ x ∈ L.val := by
+  simp only [adjoin_le_iff, Set.le_eq_subset, Set.singleton_subset_iff, SetLike.mem_coe]
+
+@[simp]
+theorem adjoin_map [IsGalois k K] (f : K →ₐ[k] K) (s : Set K) [Finite s] :
+    adjoin k (f '' s) = adjoin k s := by
+  apply val_injective; dsimp [adjoin_val]
+  rw [← IntermediateField.adjoin_map, IntermediateField.normal_map]
+
+@[simp]
+theorem adjoin_simple_map [IsGalois k K] (f : K →ₐ[k] K) (x : K) :
+    adjoin k {f x} = adjoin k {x} := by
+  simpa only [Set.image_singleton] using adjoin_map f { x }
+
+/--The `AlgEquiv` version of `adjoin_simple_map`-/
```

Comment:
```suggestion
/-- The `AlgEquiv` version of `adjoin_simple_map`. -/
```
(similarly below)

Reply:
To eliminate the `'`, I think `adjoin_simple_map_algEquiv` would be a reasonable name.